app/channel.go

```diff
@@ -670,10 +680,16 @@ func (a *App) UpdateChannelMemberRoles(channelId string, userId string, newRoles
 				member.SchemeAdmin = true
 			case schemeUserRole:
 				member.SchemeUser = true
+			case schemeGuestRole:
+				member.SchemeGuest = true
 			default:
 				// If not part of the scheme for this channel, then it is not allowed to apply it as an explicit role.
 				return nil, model.NewAppError("UpdateChannelMemberRoles", "api.channel.update_channel_member_roles.scheme_role.app_error", nil, "role_name="+roleName, http.StatusBadRequest)
 			}
+
+			if member.SchemeUser && member.SchemeGuest {
```

Comment:
I fail if you try to set a user as a User and as a Guest at the same time, actually I'm thing in the possibility of, if you are a guest, don't allow to have any SchemeUser or SchemeGuest.

Reply:
Probably makes sense to not allow to update the Scheme to guest or from guest to other thing, because, at least for now, a guest is a guest in a system wide way, so... updating the role should from/to guest would generate an inconsistent user.

Actually, I think makes sense (for the "onboarding" of the feature) to allow to "DemoteUserToGuest" (we now only have the "PromoteGuestToUser") because this way the system administrators who wants to use this feature can start using it with current existing users, not only for the new ones.

Reply:
Agree with all of this. Let's make the logic very limiting here, and we can consider relaxing it in future. At the outset, do we want the power to demote a user to guest at the channel level, or just at the system wide level?

Reply:
Done!